modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/MarketplaceStorePortlet.java

```diff
@@ -402,6 +403,11 @@ public void updateApps(
 		writeJSON(actionRequest, actionResponse, jsonObject);
 	}
 
+	@Activate
+	protected void activate() {
+		super.oAuthManager = _oAuthManager;
```

Comment:
@slnn Did you try to directly add `@Reference` to the protected field in the super class? It should work even if it's not a component?

Reply:
Hi @dantewang 

We can not do that, SF will complains:

1: @Reference should not be used in a class without @Component, see https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/check/java_module_component_check.markdown: /home/liferay/repo/portal-ce/modules/apps/marketplace/marketplace-store-web/src/main/java/com/liferay/marketplace/store/web/internal/portlet/RemoteMVCPortlet.java (SourceCheck:JavaModuleComponentCheck)

Reply:
The supper class RemoteMVCPortlet is not an abstract class.

Reply:
Hi @dantewang 

I have updated this pull to change the super class to an abstract class and then add abstract get method in it in order to make subclass to implement it.

Lily